internal/common/gamestate/gamestate.go

```diff
@@ -43,6 +43,7 @@ type GameState struct {
 // Copy returns a deep copy of the GameState.
 func (g GameState) Copy() GameState {
 	ret := g
+	ret.CommonPool = g.CommonPool
```

Comment:
Will this work for a float type?

Reply:
I think you don't need this line. CommonPool will get copied over as it should in the line ret:=g as it is a variable. This doesn't work for reference types (pointers) which is why ClientInfos etc. has to be copied over in this way.

Reply:
True, thanks Preet